common/permute_tx.c

```diff
@@ -121,7 +127,7 @@ void permute_outputs(struct bitcoin_tx_output *outputs, size_t num_outputs,
 	size_t i;
 
 	/* Now do a dumb sort (num_outputs is small). */
-	for (i = 0; i < num_outputs; i++) {
+	for (i = 0; i < num_outputs-1; i++) {
```

Comment:
This is the real fix, AFAICT.  The optimizing out the single case is unnecessary?

Reply:
And you should fix the num_inputs case as well...

Reply:
Fixed the `num_inputs` case I missed. The sanity check in `swap_inputs` and `swap_outputs` however is necessary, since `find_best_out` and `find_best_in` return a 0 offset if no better input/output is found, i.e., if the element is already at the correct location, which would then result in the overlap again.